Unstable test in master build

**Describe the bug**
Build test should consistently pass or fail 

**To Reproduce**
Create PR ( no backend changes)
`org.molgenis.emx2.graphql.TestGraphqlDatabaseFields > testCreateAndDeleteSchema FAILED
    java.lang.AssertionError: expected:<8> but was:<9>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.molgenis.emx2.graphql.TestGraphqlDatabaseFields.testCreateAndDeleteSchema(TestGraphqlDatabaseFields.java:41)
14 tests completed, 1 failed`
**Expected behavior**
Backend test should always pass or fail ( no backend code it touched in PR)

--- a/backend/molgenis-emx2-graphql/src/test/java/org/molgenis/emx2/graphql/TestGraphqlDatabaseFields.java
+++ b/backend/molgenis-emx2-graphql/src/test/java/org/molgenis/emx2/graphql/TestGraphqlDatabaseFields.java
@@ -32,16 +32,14 @@ public class TestGraphqlDatabaseFields {
 
   @Test
   public void testCreateAndDeleteSchema() throws IOException {
+    synchronized(this) {
+      var name = schemaName + Math.floor(Math.random() * 10000);
+      execute("mutation{createSchema(name:\"" + name + "\"){message}}");
+      Assert.assertTrue(database.getSchemaNames().contains(name));
 
-    // ensure schema doesn't exist
-    int realLength = database.getSchemaNames().size();
-    int length = execute("{Schemas{name}}").at("/data/Schemas").size();
-    assertEquals(realLength, length);
-    execute("mutation{createSchema(name:\"" + schemaName + "B\"){message}}");
-    Assert.assertEquals(length + 1, database.getSchemaNames().size());
-
-    execute("mutation{deleteSchema(name:\"" + schemaName + "B\"){message}}");
-    Assert.assertEquals(length, database.getSchemaNames().size());
+      execute("mutation{deleteSchema(name:\"" + name + "\"){message}}");
+      Assert.assertFalse(database.getSchemaNames().contains(name));
+    }
   }
 
   @Test

